src/NuGet.Core/NuGet.Frameworks/DualCompatibilityFramework.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Shared;
+
+namespace NuGet.Frameworks
+{
+    /// <summary>
+    /// Represents a framework that behaves as 2 potentially independent frameworks.
+    /// Ex. C++/CLI can support both .NET 5.0 and native.
+    /// This type is immutable.
+    /// </summary>
+    public class DualCompatibilityFramework : NuGetFramework
+    {
+        /// <summary>
+        /// The root framework. Any compatibility checks should be performed against this framework first.
+        /// </summary>
+        public NuGetFramework RootFramework { get; }
+
+        /// <summary>
+        /// The secondary framework. If the root framework compatibility checks fail, then the compat checks should be performed against this framework next.
+        /// </summary>
+        public NuGetFramework SecondaryFramework { get; }
+
+        private int? _hashCode;
+
+        /// <summary>
+        /// Multiple compatbility 
+        /// </summary>
+        /// <param name="framework">Root framework. Never <see langword="null"/>. </param>
+        /// <param name="secondaryFramework">Secondary framework. Never <see langword="null"/>. </param>
+        /// <exception cref="ArgumentNullException">if either <paramref name="framework"/> or <paramref name="secondaryFramework"/> are <see langword="null"/>.</exception>
+        public DualCompatibilityFramework(NuGetFramework framework, NuGetFramework secondaryFramework)
+            : base(ValidateFrameworkArgument(framework))
+        {
+            if (secondaryFramework == null)
+            {
+                throw new ArgumentNullException(nameof(secondaryFramework));
+            }
+
+            SecondaryFramework = secondaryFramework;
+            RootFramework = framework;
+        }
+
+        /// <summary>
+        /// Create a FallbackFramework from the current DualCompatibilityFramework.
+        /// </summary>
+        public FallbackFramework AsFallbackFramework()
+        {
+            return new FallbackFramework(RootFramework, new NuGetFramework[] { SecondaryFramework });
+        }
+
+        private static NuGetFramework ValidateFrameworkArgument(NuGetFramework framework)
+        {
+            if (framework == null)
+            {
+                throw new ArgumentNullException(nameof(framework));
+            }
+            return framework;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as DualCompatibilityFramework);
+        }
+
+        public bool Equals(DualCompatibilityFramework other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return Comparer.Equals(this, other)
```

Comment:
Is this supposed to be `Comparer.Equals(RootFramework, other.RootFramework)`?

Reply:
I think I did this on purpose, but your suggestion makes sense. The comparer takes NuGetFramework so it'd only compare what was getting passed. 

I think your suggestion makes more sense though.